components/dropdown/dropdown.directive.ts

```diff
@@ -205,10 +205,10 @@ export class NzDropDownDirective implements AfterViewInit, OnDestroy, OnChanges
     if (nzTrigger) {
       this.nzTrigger$.next(this.nzTrigger);
     }
-    if (nzVisible) {
+    if (nzVisible !== undefined) {
       this.inputVisible$.next(this.nzVisible);
     }
-    if (nzDisabled) {
+    if (nzDisabled !== undefined) {
```

Comment:
Here, nzVisible and nzDisabled are of type SimpleChange, not boolean. So I guess we don't need to use undefined to determine that, WDYT?

Reply:
`nzVisible` and `nzDisabled` receive **boolean** values externally, but when fetching their values from the `changes` object, there can be three possible scenarios: **undefined** (when the external value has not been changed), **true** (change to true from false), or **false** (change to false from true). However, if the code logic is simply `if(nzVisible)`, it will only process the case where it is **true**. When the external source intends to change it to **false**, it will not take effect as expected.

Reply:
I think you misunderstood that when nzDisabled is changed to false, changes.nzDisabled is of type SimpleChange, not boolean, because it's changes.nzDisabled, not this.nzDisabled.